dask/array/fft.py

```diff
@@ -24,21 +24,24 @@
     """
 
 
-def _fft_wrap(fft_func, dtype, out_chunk_fn):
-    def func(a, n=None, axis=-1):
-        if len(a.chunks[axis]) != 1:
-            raise ValueError(chunk_error % (axis, a.chunks[axis]))
+def _fft_wrap(dtype, out_chunk_fn):
```

Comment:
Can we auto-detect the dtype by calling the function on a length zero array?  Actually, we may be able to just ignore the dtype argument altogether and let `map_blocks` take care of this (it will call the function on a length one array of the input array's dtype?